src/components/baseComponent.ts

```diff
@@ -141,4 +182,29 @@ export abstract class MgtBaseComponent extends LitElement {
     });
     this.dispatchEvent(event);
   }
+
+  /**
+   * Reload the state.
+   * Returns false if already loading, unless forced.
+   * Use reload instead of load to ensure loading events are fired.
```

Comment:
Should there be a way to cancel the previous load? Or should that happen automatically if a force occurs?

Reply:
That is a great point. I was thinking about it, and I have a potential solution.

When calling `load(cancelToken)`, pass a custom `CancellationToken` object with a `isCancellationRequested: boolean` prop and `throwIfCancellationRequested(): void` function, which can be used to handle the cancellation request inside the `load()` function implementation.

It's a similar/simplified pattern from the [CancellationToken struct](https://docs.microsoft.com/en-us/dotnet/api/system.threading.cancellationtoken?view=netframework-4.8) in .NET Framework.

I guess I'm just worried that it's overkill.

@michael-hawker, @nmetulev What do you think?

Reply:
I added the pattern. Take a look and let me know what you think

Reply:
I love that we are thinking about this but in general I like to start simple and increase complexity as need arises. In this case, I'm wondering if a simple object with a `isCanceled` bool would suffice?

```ts
class CancelationToken{
  isCanceled: bool = false;
}
```

then the dev in their implementation can check the bool and return if canceled
```ts
protected loadState(cancelationToken?: CancelationToken): Promise<void> {
  await someLoading();
  if (cancelationToken.isCanceled) {
    return;
  }
  await someOtherLoading();
}
```

However, I might be oversimplifying the problem here and could be convinced otherwise :)

Reply:
Yeah, I've been out of the loop for a bit in the JavaScript world, so not sure if there's a standard best-practice or lib for this sort of paradigm.  Agree with Shane that the .NET paradigm is nice/proven at least.

I don't know how much we need to worry about other devs making components as much yet vs. what we need for just our own set of controls, so whatever makes the most sense to keep our components the simplest to maintain.

Reply:
Just ran into this api which seems to be the js way of canceling promises, with fetch supporting it out of the box: https://developer.mozilla.org/en-US/docs/Web/API/AbortController.

Reply:
The cancellationtoken implementation is great. Helps with memory leak scenarios when component has been unmounted before request completes and state gets updated.

Reply:
I'll iterate on this a bit more and submit a follow up PR. For sake of simplicity (and time) I removed the cancellation pattern for now.